pageObjects/catalogPage/fragments/catalog.list.ts

```diff
@@ -0,0 +1,41 @@
+import { normalizePrice } from '../../../helpers/helpers'
+import { Element, ElementArray } from '../../../types'
+import { CartPage } from '../../cartPage/cart.page'
+import { CatalogPage } from '../catalog.page'
+
+export class CatalogList {
+
+    private get content(): Element {
+        return $('[class*="content_type_catalog"]')
+    }
+
+    private get contentEmptyValue(): Element {
+        return this.content.$('[class="catalog-empty"]')
+    }
+
+    private get items(): ElementArray {
+        return this.content.$$('[class*="catalog-grid__cell"]')
+    }
+
+    getEmpyContentText(): Promise<string> {
+        return this.contentEmptyValue.getText()
+    }
+    
+    getItemTitle(number): Promise<string> {
+        return this.items[number - 1].$('[class="goods-tile__title"]').getText()
+    }
+
+    async getItemPrice(number): Promise<Number> {
+        const price = await this.items[number - 1].$('[class="goods-tile__price-value"]').getText()
+
+        return normalizePrice(price)
+    }
+
+    async buyItems(amount): Promise<CatalogPage> {
+        for(let i = 0; i <= (amount - 1); i++) {
+             await this.items[i].$('[class*="buy-button"]').click()
```

Comment:
not clear how this actually works
looks like a new page will open after the first click and the other iterations will be ignored, isn't it?